x/upgrade/keeper.go

```diff
@@ -169,13 +175,12 @@ func (k Keeper) TallyVotingPower(ctx sdk.Context, threshold int64) (bool, uint64
 }
 
 // GetVotingPowerThreshold returns the voting power threshold required to
-// upgrade to a new version. It converts the signal quorum parameter which
-// is a number between 0 and math.MaxUint32 representing a fraction and
-// then multiplies it by the total voting power
+// upgrade to a new version.
 func (k Keeper) GetVotingPowerThreshold(ctx sdk.Context) sdkmath.Int {
-	quorum := k.SignalQuorum(ctx)
+	// contract: totalVotingPower should not exceed MaxUit64
 	totalVotingPower := k.stakingKeeper.GetLastTotalPower(ctx)
-	return quorum.MulInt(totalVotingPower).RoundInt()
+	thresholdFraction := SignalThreshold(ctx.BlockHeader().Version.App)
+	return totalVotingPower.MulRaw(thresholdFraction.Numerator).QuoRaw(thresholdFraction.Denominator)
```

Comment:
We may want to investigate / enforce that this never overflows

Reply:
ref: https://github.com/celestiaorg/celestia-app/issues/2878